app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php

```diff
@@ -123,34 +123,42 @@ private function getEmailNotification()
      */
     public function execute()
     {
-        /** @var \Magento\Framework\Controller\Result\Json $resultJson */
-        $resultJson = $this->resultJsonFactory->create();
+        try {
+            /** @var \Magento\Framework\Controller\Result\Json $resultJson */
+            $resultJson = $this->resultJsonFactory->create();
 
-        $postItems = $this->getRequest()->getParam('items', []);
-        if (!($this->getRequest()->getParam('isAjax') && count($postItems))) {
-            return $resultJson->setData([
-                'messages' => [__('Please correct the data sent.')],
-                'error' => true,
-            ]);
-        }
+            $postItems = $this->getRequest()->getParam('items', []);
+            if (!($this->getRequest()->getParam('isAjax') && count($postItems))) {
+                return $resultJson->setData([
+                    'messages' => [__('Please correct the data sent.')],
+                    'error' => true,
+                ]);
+            }
+
+            foreach (array_keys($postItems) as $customerId) {
+                $this->setCustomer($this->customerRepository->getById($customerId));
+                $currentCustomer = clone $this->getCustomer();
 
-        foreach (array_keys($postItems) as $customerId) {
-            $this->setCustomer($this->customerRepository->getById($customerId));
-            $currentCustomer = clone $this->getCustomer();
+                if ($this->getCustomer()->getDefaultBilling()) {
+                    $this->updateDefaultBilling($this->getData($postItems[$customerId]));
+                }
+                $this->updateCustomer($this->getData($postItems[$customerId], true));
+                $this->saveCustomer($this->getCustomer());
 
-            if ($this->getCustomer()->getDefaultBilling()) {
-                $this->updateDefaultBilling($this->getData($postItems[$customerId]));
+                $this->getEmailNotification()->credentialsChanged($this->getCustomer(), $currentCustomer->getEmail());
             }
-            $this->updateCustomer($this->getData($postItems[$customerId], true));
-            $this->saveCustomer($this->getCustomer());
 
-            $this->getEmailNotification()->credentialsChanged($this->getCustomer(), $currentCustomer->getEmail());
+            return $resultJson->setData([
+                'messages' => $this->getErrorMessages(),
+                'error' => $this->isErrorExists()
+            ]);
+        } catch (\Exception $e) {
```

Comment:
This behavior should be adjusted slightly.
We shouldn't send details about all exceptions to the frontend since some details might be overwhelming and confusing for non-technical persons. Usually, we show information about LocalizedException and similar.
If there's some catchable fatal error, a merchant should be informed, literally, that "Something went wrong". All other details should be logged into an error log.